plugin/core/views.py

```diff
@@ -976,44 +975,36 @@ def _format_diagnostic_related_info(
     location = info["location"]
     return '<a href="{}">{}</a>: {}'.format(
         location_to_href(config, location),
-        location_to_human_readable(config, base_dir, location),
-        info["message"]
+        text2html(location_to_human_readable(config, base_dir, location)),
+        text2html(info["message"])
     )
 
 
-def _with_color(text: Any, hexcolor: str) -> str:
-    return '<span style="color: {};">{}</span>'.format(hexcolor, text)
+def _with_color(text: str, hexcolor: str) -> str:
+    return '<span style="color: {};">{}</span>'.format(hexcolor, text2html(text))
```

Comment:
I would not place the text2html inside this function, 
To avoid unnecessary work, for example
```py
formatted.append(_with_color("(", "color(var(--foreground) alpha(0.6))")) // we would unnecessary call text2html
formatted.append(_with_color(str(code), "color(var(--foreground) alpha(0.6))")) // only here calling text2html would be appreciated
formatted.append(_with_color(")", "color(var(--foreground) alpha(0.6))")) // we would unnecessary call text2html
```

Instead I would suggest:
```py
formatted.append(_with_color("(", "color(var(--foreground) alpha(0.6))"))
formatted.append(_with_color(text2html(code), "color(var(--foreground) alpha(0.6))"))
formatted.append(_with_color(")", "color(var(--foreground) alpha(0.6))")) 
```